docs/practices/style.md

```diff
@@ -407,16 +407,49 @@ find a particular span in logs or other tools ingesting the span data. If a
 span begins at the top of a function, prefer giving it a name of that function,
 otherwise prefer a `snake_case` name.
 
-Use the regular span API over convenience macros such as `#[instrument]`, as
-this allows instrumenting portions of a function without affecting the code
-structure:
+Default to the [`#[tracing::instrument]`](instrument) macro. When instrumenting asynchronous
+functions either this or the `Future::instrument` is **required**. The spans produced in these
+contexts will be wrong if the span is held across an await point. Correct handling of this is
+handled for you via either of the suggested methods. If instrumentation of async functions is not
+done according to this guidance it can lead to difficult to troubleshoot issues such as stack
+overflows.
+
+Always use `level`, `target` and `skip_all` to avoid implicit defaults for the level, target or
```

Comment:
Always explicitly specify the `level`, `target`, and `skip_all` fields and do not rely on the default values.  This avoids adding all function arguments as span `field`s which can lead recording potentially unnecessary and expensive information.  Carefully consider which information needs recording and the cost of recording the information when using the `fields` option.